matching-service/controller/matching-controller.js

```diff
@@ -0,0 +1,258 @@
+import Redis from 'ioredis';
+import { io } from '../index.js';
+import {fetchCategories} from '../internal-services/question-service.js';
+
+
+const redis = new Redis({
+    host: 'localhost',
+    port: 6379,
+  });
+
+const CHECK_INTERVAL = 1000; // interval to run schedule matching
+const LOOSEN_DIFFICULTY_TIME = 10000; // number of checks by 
+const TIMEOUT = 30000;
+
+const cancelMatchmakeUsers = new Map();
```

Comment:
I notice there's no action to remove the entries. Prob not too critical now. Just concerns of memory overload

Reply:
The matchUserInQueue will keep popping all the user from the queue, so eventually all the user will be popped from the queue.

Reply:
Actually I meant the hashmap, but should be no worries for now